### ISSUE TITLE
When polybar starts with WM, workspace list is empty

### ISSUE THREAD

**Issue by jonhoo** [polybar/polybar#1915]
## Describe the issue
I'm using xmonad, and starting `polybar` directly from xmonad when it starts up. This used to work fine, but recently (I suspect after https://github.com/polybar/polybar/pull/882) my list of workspaces is now shown as entirely empty. If I restart `polybar` shortly after, the workspaces appear as normal.

**Was it working before?**
I never experienced this issue with my previous build, which was gfe7dc28b, but am seeing it now with g2b31a3f1.

## To Reproduce
It is surprisingly hard to reproduce this, as it only happens very occasionally at boot. I _believe_ this config is sufficient to reproduce, but have yet to confirm. Otherwise, my full config is [this one](https://github.com/jonhoo/configs/blob/b6d4aee616d234f581766b467a0e772e775cc53f/gui/.config/polybar/config). Note that I _do_ overwrite `monitor`, but not sure whether that would affect any of this.
```dosini
[bar/main]
monitor = ${env:MONITOR:eDP-1}
modules-left = ewmh

[module/ewmh]
type = internal/xworkspaces
pin-workspaces = false

[settings]
screenchange-reload = true
```

Essentially, have your WM run `polybar` on boot with:
```console
$ env "MONITOR=$m" polybar --reload main
```
where `$m` is the monitor you want to use (again, not sure if the monitor part is relevant, but including it for completeness). Reboot and observe the bar. Every few boots, the list of workspaces will be empty. If you then restart polybar, it will be populated again.

## Polybar Log

I forgot to copy the log last time this happened, and am having a hard time reproducing it reliably. Will post the log next time it happens.

## Environment:
* WM: xmonad
* Distro: Arch Linux
* Output of `polybar -vvv`:
```
polybar 3.4.0-24-g2b31a3f1

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Wall -Wextra -Wpedantic -O3 -DNDEBUG
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
```

**Comment by kronn** [comment]
Just one question pops into my mind: Do the workspaces show up if you switch the workspace? If so, then the first batch of events is not processed.

**Comment by jonhoo** [comment]
No, I don't think so, but will make sure to check next time this happens!

**Comment by wedens** [comment]
It seems to happen when polybar starts before xmonad.

I can consistently reproduce it, as it seems that in my configuration polybar as systemd service always starts before WM.

Same workaround with restarting polybar from within xmonad (when it's known that it's already running) works.

**Comment by TaylanTatli** [comment]
```
polybar|error: Disabling module "ewmh" (reason: The WM does not support _NET_CURRENT_DESKTOP, aborting...)
```
Same problem. Using openbox, and it was working, now i can't see workspace list.

**Comment by patrick96** [comment]
I'm not really sure why this only starts happening now for everyone. That requirement (support for _NET_CURRENT_DESKTOP) has been part of the module since it was created four years ago.
